src/components/assetlib_projects/asset_library_projects.gd

```diff
@@ -0,0 +1,353 @@
+		if _current_assets:
+			_current_page = min(_current_assets.pages - 1, _current_page + 1)
+		else:
+			_current_page = 0
+	)
+	_last_button.pressed.connect(func():
+		_current_page = _current_assets.pages - 1 if _current_assets else 0
+	)
+	
+	await _setup_version_button()
+	
+	_fetch_assets()
+
+
+func _setup_version_button():
+	var versions = await _fetch_versions()
+	versions.reverse()
+	if versions == []:
+		_version_option.disabled = true
+		return
+	
+	for version in versions:
+		_version_option.add_item(version)
+
+
+func _on_search_field_text_changed(_new_text: String):
+	_last_text_edit = Time.get_ticks_msec()
+	await get_tree().create_timer(1).timeout
+	if Time.get_ticks_msec() - _last_text_edit >= 1000:
+		_fetch_assets()
+
+
+func _fetch_assets(_trash = null, reset_page_number: bool = true):
+	if not _fetched_versions:
+		return
+	if reset_page_number:
+		_internal_current_page = 0
+	var query = _generate_query(_generate_query_dictionary())
+	_asset_querier.cancel_request()
+	_asset_querier.request(query)
+	_message(false, tr("Fetching assets…"))
+
+
+func _on_asset_querier_request_completed(result: int, response_code: int,
+		_headers: PackedStringArray, byte_body: PackedByteArray):
+	if result != HTTPRequest.RESULT_SUCCESS or response_code != 200:
+		_message(true, tr("Fetching assets failed."))
+		_current_assets = {}
+		return
+	
+	var body = byte_body.get_string_from_ascii()
+	var json = JSON.new()
+	if json.parse(body) != OK or not json.data is Dictionary:
+		_message(true, tr("Received unexpected data."))
+		_current_assets = {}
+		return
+	_current_assets = json.data
+	_clear_message()
+
+
+func _display_assets(current_assets: Dictionary):
+	_current_assets = current_assets
+	_clear_asset_display()
+	if current_assets == {} or current_assets.total_items == 0:
+		return
+	
+	_display_navigation()
+	
+	var assets = current_assets.result
+	for asset_data in assets:
+		var asset = _ASSET_LISTING.instantiate().init_asset_listing_interactible(
+				int(asset_data.asset_id),
+				asset_data.title,
+				asset_data.category,
+				asset_data.author,
+				asset_data.cost,
+				projects
+		)
+		
+		_asset_list.add_child(asset)
+
+
+func _clear_navigation_buttons():
+	for button in _page_buttons.get_children():
+		button.hide()
+
+
+## Sets up the navigation buttons, taking into account number of pages
+## and current page number.
+func _display_navigation():
+	_clear_navigation_buttons()
+	if not _current_assets or _current_assets.pages == 0:
+		_navigation_buttons.hide()
+		return
+	_navigation_buttons.show()
+	var buttons = _page_buttons.get_children()
+	@warning_ignore("integer_division")
+	var center_offset = _MAX_PAGE_BUTTONS / 2
+	var start_page = max(0, _current_page - center_offset)
+	for i in _MAX_PAGE_BUTTONS:
+		var page = start_page + i
+		if page > _current_assets.pages - 1:
+			break
+		buttons[i].show()
+		buttons[i].text = str(start_page + i + 1)
+		buttons[i].disabled = start_page + i == _current_page
```

Comment:
This doesn't always display 10 pages (it display less when one approaches the end).  I initially considered it a bug, but then noticed Godot's official project manager also displays that behaviour and have therefore made no effor to fix it.